src/iqm/cirq_iqm/devices/iqm_device_metadata.py

```diff
@@ -35,6 +35,7 @@ class IQMDeviceMetadata(devices.DeviceMetadata):
         operations: Supported quantum operations of the device, mapping op types to their possible loci.
         gateset: Native gateset of the device. If None, a default IQM device gateset will be used.
         resonators: computational resonators of the device
+        architecture: architecture from which values of the other arguments were obtained
     """
 
     QUBIT_NAME_PREFIX: str = 'QB'
```

Comment:
Is this still needed with DQA?

Reply:
It's still needed because it's used in `IQMDevice.get_qubit_index` and `IQMDeviceMetadata.from_qubit_indices`.

* `get_qubit_index` is not used anywhere in code. But could be useful for some purpose, not sure if there was some requirement to have it?

* `from_qubit_indices` makes creating the hardcoded devices like Apollo a bit nicer

Both are probably something we could get rid of if we really want, but I would still keep this and not change these things in this PR.